File: app/src/main/java/com/dungnd/android23/buoi4bai3/FolderAdapter.java
package com.dungnd.android23.buoi4bai3;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.dungnd.android23.R;

import java.util.List;

public class FolderAdapter extends RecyclerView.Adapter<FolderAdapter.FolderViewHolder> {

    private List<Folder> mListFolder;

    public FolderAdapter(List<Folder> mListFolder) {
        this.mListFolder = mListFolder;
    }

    @NonNull
    @Override
    public FolderViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view  = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_folder,parent,false);

        return new FolderViewHolder(view) ;
    }

    @Override
    public void onBindViewHolder(@NonNull FolderViewHolder holder, int position) {
        Folder folder = mListFolder.get(position);
        if (folder == null){
            return;
        }
        holder.tvName.setText(folder.getTv_name());
    }


    @Override
    public int getItemCount() {

        return mListFolder != null ? mListFolder.size() : 0;
    }

    protected class FolderViewHolder extends RecyclerView.ViewHolder{
        private TextView tvName;

        public FolderViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tv_name);

        }
    }
}

File: tunanh_bai7/src/main/java/com/example/tuananh/FolderAdapter.kt
package com.example.tuananh

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class FolderAdapter(var context : Context, var array : ArrayList<FolderModel>) : RecyclerView.Adapter<FolderAdapter.FolderViewHolder>() {
//    private var fdList:ArrayList<FolderModel> = ArrayList()
//    private var onClickITem :((FolderModel)-> Unit)?=null
//    fun additem(item:ArrayList<FolderModel>){
//        this.array=item
//        notifyDataSetChanged()
//    }
    var onItemClick: ((FolderModel, Int) -> Unit)? = null
    class FolderViewHolder(v:View):RecyclerView.ViewHolder(v){
        private var title=v.findViewById<TextView>(R.id.title_item)
        private var content=v.findViewById<TextView>(R.id.content_item)
        fun bindView(folderModel: FolderModel){
            title.text=folderModel.title
            content.text=folderModel.title

        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FolderViewHolder = FolderViewHolder(
        LayoutInflater.from(parent.context).inflate(R.layout.item_folder,parent,false))

    override fun onBindViewHolder(holder: FolderViewHolder, position: Int) {
        val fd=array.get(position)
        holder.bindView(fd)
        holder.itemView.setOnClickListener {
            onItemClick?.invoke(array.get(position), position)
        }

//        holder.setOnClickListener{onClickITem?.invoke(fd)}
//        setonClickitem(onClickITem?.invoke(fd),position)
    }

    override fun getItemCount(): Int {
        return array.size
    }
}